### ISSUE TITLE
Add test for setting `Programmed` status condition in `HTTPRouteReconciler`

### ISSUE THREAD

**Issue by pmalek** [Kong/kubernetes-ingress-controller#3793]
This issue tracks the effort to add a test that checks that `Programmed` status condition is added properly in `HTTPRouteReconciler`.

### Acceptance criteria

- [ ] `Programmed` status condition is being checked for existence and correctness in `HTTPRouteReconciler` tests (either regular unit or `envtest` based tests)

**Comment by czeslavo** [comment]
The envtest added in https://github.com/Kong/kubernetes-ingress-controller/pull/4267 seems to be satisfying this issue ACs, @pmalek do you think we can close it?

**Comment by pmalek** [comment]
It's sort of implicit don't you think? Ideally we'd have a dedicated test (or a test suite testing various things about HTTPRoute) for this.

**Comment by czeslavo** [comment]
Yeah, agree. We could check the observed generation is properly set. Anyways, `TestHTTPRouteReconciliation_DoesNotBlockSyncLoopWhenStatusQueueBufferIsExceeded` might be a good starting point for anyone starting working on this issue.
